File: linuxcontainer/go/ns_uts.go
package main

import (
	"fmt"
	"syscall"
	"unsafe"
)

func main() {
	r1, _, err := syscall.RawSyscall6(syscall.SYS_CLONE,
		uintptr(syscall.CLONE_NEWUTS)|uintptr(syscall.SIGCHLD), 0, 0, 0, 0, 0)
	if err != 0 {
		fmt.Println(err)
		return
	}
	fmt.Printf("ok %d\n", int(r1))
	if int(r1) == 0 {
		syscall.Wait4(int(r1), nil, 0, nil)
		return
	}
	syscall.Sethostname([]byte("NewHostname"))

	var argv = []string{
		"/bin/bash",
	}

	if argvp, err := syscall.SlicePtrFromStrings(argv); err == nil {
		_, _, err = syscall.RawSyscall(syscall.SYS_EXECVE,
			uintptr(unsafe.Pointer(argvp[0])),
			uintptr(unsafe.Pointer(&argvp[0])),
			0)
	}
}

File: linuxcontainer/c/ns_uts.c
// Test UTS namespace

#define _GNU_SOURCE
#include <stdio.h>
#include <sched.h>
#include <signal.h>
#include <unistd.h>

#define STACK_SIZE (1024 * 1024)
static char child_stack[STACK_SIZE];
char * const child_args[] = {
  "/bin/bash",
  NULL,
};

int child_main(void *arg)
{
        printf("world\n");
	// set hostname in my namespace
        sethostname("in namespace", 12);
        execv(child_args[0], child_args);
        printf("oops\n");
        return 1;
}

int main(int argc, char **argv)
{
        int child_pid;
        printf("hello\n");
        child_pid = clone(child_main, child_stack + STACK_SIZE, SIGCHLD | CLONE_NEWUTS, NULL);
        waitpid(child_pid, NULL, 0);
        return 0;
}

  
